momentum/4/lua/ref-msys-tls-params-set.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "msys.tls_params.set"
+description: "msys tls_params set Set a tls parameter string in the Momentum configuration, overriding if the parameter is already set"
+---
+
+<a name="lua.ref.msys.tls_params.set"></a> 
+## Name
+
+msys.tls_params.set — Set a tls parameter string in the Momentum configuration, overriding if the parameter is already set.
+
+## Synopsis
+
+`require('msys.tls_params');`
+
+`msys.tls_params.set(tls_params, "param", "val");`
+
+```
+tls_params: mixed
+param: string
+val: string
+```
+
+## Description
+
+This function sets a tls parameter string in the Momentum configuration, overriding if the parameter is already set.
+
+This should only be used in hooks where the tls parameters are passed as an argument such as `get_outbound_tls_parameters` and `get_inbound_tls_parameters`.
+
+<a name="lua.ref.msys.tls_params.set.example"></a> 
+
+The following example shows how to use this function in the `get_outbound_tls_parameters` hook to override all outbound tls parameters in a mutual tls connection.
+It is important to note that the use of this hook will reset outbound tls config for the current connection, requiring the user to set a value for all keys listed in the example below.
```

Comment:
I'd just say that it will override the values set in config rather than all are required - but not a huge deal.

Reply:
I'll adjust this.